PennMobile/Dining/Controllers/DiningDollarsTransactionViewController.swift

```diff
@@ -7,3 +7,98 @@
 //
 
 import Foundation
+
+class DiningDollarsTransactionViewController: GenericTableViewController, Requestable {
+    
+    let transactionUrl = "https://api.pennlabs.org/dining/transactions"
+    var transactionHistory: [Transaction]?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.title = "Transaction History"
+
+        tableView.delegate = self
+        tableView.dataSource = self
+                    
+        tableView.register(TransactionTableViewCell.self, forCellReuseIdentifier: TransactionTableViewCell.identifier)
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        
+        PennCashNetworkManager.instance.getTransactionHistory { data in
+            if let data = data, let str = String(bytes: data, encoding: .utf8) {
+                UserDefaults.standard.setLastTransactionRequest()
+                UserDBManager.shared.saveTransactionData(csvStr: str) {
+                    self.fetchTransactionData { (results, error) in
+                        DispatchQueue.main.async {
+                            self.transactionHistory = results
+                            self.tableView.reloadData()
+                        }
+                    }
+                }
+            } else {
+                self.fetchTransactionData { (results, error) in
+                    DispatchQueue.main.async {
+                        self.transactionHistory = results
+                        self.tableView.reloadData()
+                    }
+                }
+            }
+        }
+    }
+}
+
+// MARK: - Table View Functions
+extension DiningDollarsTransactionViewController {
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return transactionHistory?.count ?? 0
+    }
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: TransactionTableViewCell.identifier, for: indexPath) as! TransactionTableViewCell
+        let transaction: Transaction = transactionHistory![indexPath.item]
```

Comment:
Avoid force unwraps like this, make this handle the optional case with an empty cell

Reply:
This is perfectly safe and a rare instance where the bang operator is appropriate. If ```transactionHistory``` is null, then there will be zero cells by the numberOfCells delegate method and this delegate method will never be called.